src/login/LoginViewModel.ts

```diff
@@ -345,14 +347,19 @@ export class LoginViewModel implements ILoginViewModel {
 					await this._credentialsProvider.store(newCredentials)
 				} catch (e) {
 					if (e instanceof KeyPermanentlyInvalidatedError) {
-						await this._credentialsProvider.clearCredentials()
+						await this._credentialsProvider.clearCredentials(e)
 						await this._updateCachedCredentials()
+					} else if (e instanceof DeviceStorageUnavailableError) {
```

Comment:
Why do we need a catch here and a catch outside? If one of them fails with this error, all others will probably fail as well

Reply:
this preserves the behaviour we had previously: 
* when deleting duplicate credentials fails, immediately bail out of the login process
* when saving credentials fails with a known error, still log in, just skip the saving step

both cases should probably log something, but the second one must be caught in the inner `try` so the login can proceed.

Reply:
Right, it's about deletion, not storing. I would suggest to then `break` there

Reply:
wait, sorry, github is confusing, it's fine